src/esm_runscripts/filedicts.py

```diff
@@ -341,6 +358,83 @@ def _path_type(self, path: pathlib.Path) -> Union[str, bool]:
+            error_text = (
+                f"{error_text}"
+                f"- the ``type`` variable is missing. Please define a ``type`` "
+                f"({types_text})\n"
+            )
+            missing_vars = (
+                f"{missing_vars}    ``type``: forcing/input/restart/outdata/...\n"
+            )
+        else:
+            if self["type"] not in self.all_model_filetypes:
+                error_text = (
+                    f"{error_text}"
+                    f"- ``{self['type']}`` is not a supported ``type`` "
+                    f"(``files.{self.name}.type``), please choose one of the following "
+                    f"types: {types_text}\n"
+                )
+                self._original_filedict["type"] = f"``{self._original_filedict['type']}``"
+
+        missing_vars = (
+            f"Please, complete/correct the following vars for your file:\n\n"
+            f"{self.pretty_original_filedict()}"
+            f"{missing_vars}"
+        )
+        if "path_in_computer" not in self.keys() and self.get("type") in input_file_types:
+            error_text = (
+                f"{error_text}"
+                f"- the ``path_in_computer`` variable is missing. Please define a "
+                f"``path_in_computer`` (i.e. the path to the file excluding its name)."
+                f" NOTE: this is only required for {', '.join(input_file_types)} file "
+                f"types\n"
+            )
+            missing_vars = (
+                f"{missing_vars}    ``path_in_computer``: <path_to_file_dir>\n"
+            )
+
+        if "name_in_computer" not in self.keys() and self.get("type") in input_file_types:
+            error_text = (
+                f"{error_text}"
+                f"- the ``name_in_computer`` variable is missing. Please define a ``name_in_computer`` "
+                f"(i.e. name of the file in the work folder). NOTE: this is only required for "
+                f"{', '.join(input_file_types)} file types\n"
+            )
+            missing_vars = (
+                f"{missing_vars}    ``name_in_computer``: <name_of_file_in_computer_dir>\n"
+            )
+
+        if "name_in_work" not in self.keys() and self.get("type") in output_file_types:
+            error_text = (
+                f"{error_text}"
+                f"- the ``name_in_work`` variable is missing. Please define a ``name_in_work`` "
+                f"(i.e. name of the file in the work folder). NOTE: this is only required for "
+                f"{', '.join(output_file_types)} file types\n"
+            )
+            missing_vars = (
+                f"{missing_vars}    ``name_in_work``: <name_of_file_in_work_dir>\n"
+            )
+
+        if error_text:
+            error_text = (
+                f"The file dictionary ``{self.name}`` is missing relevant information "
+                f"or is incorrect:\n{error_text}"
+            )
+            user_error("File Dictionaries", f"{error_text}\n{missing_vars}")
```

Comment:
Is this the correct indentation?

Reply:
Sorry my bad. It looks more clear when you view it in the editor instead of GitHub